### ISSUE TITLE
Some highlighted rendering improvement suggestions

### ISSUE THREAD

**Issue by mylovesaber** [BashSupport-Pro/bashsupport-pro#116]
The plugin works great and I've paid the annual fee. But I have encountered several issues while using and am confused on how to solve.

---

![image](https://user-images.githubusercontent.com/36480362/202393109-4c53dc7b-3c55-4a5e-89ef-82d7cb2f3c90.png)
As shown in the picture:
Variables will have a highlight effect (different colors under different themes), but if they are wrapped in braces, the highlight effect will be lost. Is it possible to make variables wrapped in braces also have a highlight effect?
For shell variables, whether to use braces or not has no effect on the meaning of the variable itself, but variables wrapped in braces are very convenient when modifying names in batches
Example:
Original variable names aaa and aaabbb
After wrapping in curly braces, modifying 
${aaa} 
to 
${ccc} 
in batches will not change the variable name of 
${aaabbb} 
to 
${cccbbb}, 
but if not wrapping in curly braces, $aaabbb will become $cccbbb

---

![image](https://user-images.githubusercontent.com/36480362/202402049-af0c8714-3f54-4ea4-98a9-c6c62034d4bc.png)
How to make array variables created by mapfile be local in function? I haven't found a solution to remove this squiggly line prompt.

---

![image](https://user-images.githubusercontent.com/36480362/202404683-c0195dcc-d503-407e-aa68-a58f535d82be.png)
Is it possible to add these structures to the prompt?

**Comment by jansorg** [comment]
Thank you for reporting this!

1. Highlighting of parameter expansions, e.g. `${HOME}`. 
This is supposed to be highlighted in the same way as plain variable references (e.g. `$HOME`), configured at `Settings > Editor > Color Scheme > BashSupport Pro > Variables > Composed variable`. 
This is a bug and I'll fix it for the next update of 3.0.

2. By default, the declared variable is global. You can make the variable local by using `local SYNC_NAME_OPERATION_LIST` first, e.g. 
   ```bash
   local SYNC_NAME_OPERATION_LIST
   # ...
   mapfile -t SYNC_NAME_OPERATION_LIST # ...
   ```
   The alternative is to just reference the variable, but it still would be global
    If you don't like the "variable could be local" messages, you can disable the inspection in the submenu when you press ALT+ENTER on the message (or use the light bulb)
3. That would be new live templates. I'm trying to only show the most useful live templates in the suggestions. I've created https://github.com/BashSupport-Pro/bashsupport-pro/issues/117 to track this issue. For now, you could create your own live templates for these suggestions, as shown in the screenshot.
![image](https://user-images.githubusercontent.com/11473063/202416836-f3e231f5-f626-45f8-a26d-0477287520e2.png)

**Comment by mylovesaber** [comment]
I found a strange issue, might be a bug. Here is the source code of the problematic shell script:
```shell
#!/bin/bash
SyncLocateFiles(){
    days=$(( 1 - 1 ))
}
```
![image](https://user-images.githubusercontent.com/36480362/202484094-29f5c5ae-118b-4b5d-8f0a-4fd8c09647b6.png)
Looks totally normal, right? 
If I press the Enter key on my keyboard at the end of the line with the function name, where the blue cursor is in the screenshot, it will look like this:
![image](https://user-images.githubusercontent.com/36480362/202485120-7beca810-2257-476e-baa2-4a630f45b424.png)
I've tested different code modules and it seems that this problem only occurs when using double parentheses for arithmetic operations.

**Comment by jansorg** [comment]
> I found a strange issue, might be a bug. Here is the source code of the problematic shell script:

Thanks! Good observation, that's a bug :) I'm tracking this at https://github.com/BashSupport-Pro/bashsupport-pro/issues/118 and already fixed it for the next update.

**Comment by jansorg** [comment]
This is fixed by 3.0.3, which is now available. Please create new issues if there's anything left to do or if there's another problem. Thanks!
